### PR TITLE
Fixes #2037: apoc.export.cypher.all to generate constraints with IF NOT EXISTS

### LINKED ISSUE #2037: apoc.export.cypher.all to generate constraints with IF NOT EXISTS
## Feature description (Mandatory)
Add an option (or make default) the generation of IF NOT EXISTS when outputting the cypher to generate the constraints.
Currently if the loading of the cypher file is interrupted it requires to remove the constraints to be running without error.

## Considered alternatives
N/A

## How this feature can improve the project?
This allows to re-run a generated cypher (in case it did not complete) specially if MERGE is the strategy used.

### PR DESCRIPTION
**Pull request by vga91**
Fixes neo4j-contrib#2037
Fixes neo4j-contrib#2073

Changed index creation with new syntax

### CODE REVIEW

File: core/src/main/java/apoc/export/util/ExportConfig.java [hunk 1]
```diff
@@ -97,6 +98,7 @@ public ExportConfig(Map<String,Object> config) {
         this.config = config;
         this.streamStatements = toBoolean(config.get("streamStatements")) || toBoolean(config.get("stream"));
         this.writeNodeProperties = toBoolean(config.get("writeNodeProperties"));
+        this.ifNotExists = toBoolean(config.get("ifNotExists"));
```
Comment (jexp) [inline core/src/main/java/apoc/export/util/ExportConfig.java:101]: should the default be true? so it's more idempotent

Comment (jexp) [reply core/src/main/java/apoc/export/util/ExportConfig.java:101]: and then folks that have older versions can disable it.
or we introduce it here with default false and then change it in the next release (4.4)?

**Review by jexp — APPROVED** [review]

**Comment by havardthom** [comment]
Can this also be merged to 4.4?

**Comment by conker84** [comment]
@havardthom done: https://github.com/neo4j-contrib/neo4j-apoc-procedures/pull/2512

**Pull request merged**